tvbwidgets/ui/ts_widget.py

```diff
@@ -545,6 +587,210 @@ def _update_fig(self):
+
+    def _create_channel_selection_area(self, array_wrapper, no_checkbox_columns=2):
+        # type: (ABCDataWrapper) -> widgets.Accordion
+        """ Create the whole channel selection area: Submit button to update plot, Select/Uselect all btns,
+            State var. & Mode selection and Channel checkboxes
+        """
+        # checkboxes
+        checkboxes_region = self._create_checkboxes(array_wrapper=array_wrapper,
+                                                    no_checkbox_columns=no_checkbox_columns)
+        for cb_stack in checkboxes_region.children:
+            cb_stack.layout = widgets.Layout(width='50%')
+
+        # selection submit button
+        self.submit_selection_btn = widgets.Button(description='Submit selection', layout=self.BUTTON_STYLE)
+        self.submit_selection_btn.on_click(self._update_ts)
+
+        # select/unselect all buttons
+        select_all_btn, unselect_all_btn = self._create_select_unselect_all_buttons()
+
+        # select dimensions buttons (state var. & mode)
+        selections = self._create_dim_selection_buttons(array_wrapper=array_wrapper)
+        for selection in selections:
+            selection.layout = widgets.Layout(width='50%')
+
+        # add all buttons to channel selection area
+        channels_region = widgets.VBox(children=[self.submit_selection_btn, widgets.HBox(selections),
+                                                 widgets.HBox([select_all_btn, unselect_all_btn]),
+                                                 checkboxes_region])
+        channels_area = widgets.Accordion(children=[channels_region], selected_index=None,
+                                          layout=widgets.Layout(width='70%'))
+        channels_area.set_title(0, 'Channels')
+        return channels_area
+
+    def _update_ts(self, btn):
+        self.logger.debug('Updating TS')
+        ch_names = list(self.ch_names)
+
+        # save selected channels using their index in the ch_names list
+        picks = []
+        for cb in list(self.checkboxes.values()):
+            ch_index = ch_names.index(cb.description)  # get the channel index
+            if cb.value:
+                picks.append(ch_index)  # list with number representation of channels
+
+        # if unselect all
+        # TODO: should we remove just the traces and leave the channel names and the ticks??
+        if not picks:
+            self.fig.data = []  # remove traces
+            self.fig.layout.annotations = []  # remove channel names
+            self.fig.layout.yaxis.tickvals = []  # remove ticks between channel names and traces
+            return
+
+        # get data and names for selected channels; self.raw is updated before redrawing starts
+        data, _ = self.raw[:, :]
+        data = data[picks, :]
+        ch_names = [ch_names[i] for i in picks]
+
+        # redraw the entire plot
+        self.plot_ts_with_plotly(data, ch_names)
+
+
```

Comment:
This file has become large (to read/manipulate). 
Would it be worthy to split it in 2 or 3 (maybe in its own `ts` folder?

Reply:
Yup, I agree. I will create a dedicated TS folder for the TS widget and break this file into multiple modules.